cmd/vicadmin/vicadm.go

```diff
@@ -135,51 +142,55 @@ func logFiles() []string {
 	return names
 }
 
-func configureReaders() []entryReader {
+func configureReaders() map[string]entryReader {
 	defer trace.End(trace.Begin(""))
 
-	pprofPaths := []string{
+	pprofPaths := map[string]string{
 		// verbose
-		"/debug/pprof/goroutine?debug=2",
+		"verbose": "/debug/pprof/goroutine?debug=2",
 		// concise
-		"/debug/pprof/goroutine?debug=1",
-		"/debug/pprof/block?debug=1",
-		"/debug/pprof/heap?debug=1",
-	}
-
-	pprofSources := []string{
-		// vch-init
-		"http://127.0.0.1:6060",
-		// TODO: add pprof for the other components
-	}
-
-	readers := []entryReader{
-		fileReader("/proc/mounts"),
-		commandReader("uptime"),
-		commandReader("df"),
-		commandReader("free"),
-		commandReader("netstat -ant"),
-		commandReader("sudo iptables --list"),
-		commandReader("ip link"),
-		commandReader("ip addr"),
-		commandReader("ip route"),
-		commandReader("lsmod"),
+		"concise": "/debug/pprof/goroutine?debug=1",
+		"block":   "/debug/pprof/block?debug=1",
+		"heap":    "/debug/pprof/heap?debug=1",
+	}
+
+	pprofSources := map[string]string{
+		"docker":    pprof.GetPprofEndpoint(pprof.DockerPort).String(),
+		"imagec":    pprof.GetPprofEndpoint(pprof.ImagecPort).String(),
+		"portlayer": pprof.GetPprofEndpoint(pprof.PortlayerPort).String(),
+		"vicadm":    pprof.GetPprofEndpoint(pprof.VicadminPort).String(),
+		"vic-init":  pprof.GetPprofEndpoint(pprof.VchInitPort).String(),
+	}
+
+	readers := map[string]entryReader{
+		"proc-mounts": fileReader("/proc/mounts"),
+		"uptime":      commandReader("uptime"),
+		"df":          commandReader("df"),
+		"free":        commandReader("free"),
+		"netstat":     commandReader("netstat -ant"),
+		"iptables":    commandReader("sudo iptables --list"),
+		"ip-link":     commandReader("ip link"),
+		"ip-addr":     commandReader("ip addr"),
+		"ip-route":    commandReader("ip route"),
+		"lsmod":       commandReader("lsmod"),
 		// TODO: ls without shelling out
-		commandReader("ls -l /dev/disk/by-path"),
-		commandReader("ls -l /dev/disk/by-label"),
+		"disk-by-path":  commandReader("ls -l /dev/disk/by-path"),
+		"disk-by-label": commandReader("ls -l /dev/disk/by-label"),
 		// To check we are not leaking any fds
-		commandReader("ls -l /proc/self/fd"),
+		"proc-self-fd": commandReader("ls -l /proc/self/fd"),
 	}
 
 	// add the pprof collection
-	for _, source := range pprofSources {
-		for _, paths := range pprofPaths {
-			readers = append(readers, urlReader(source+paths))
+	for sname, source := range pprofSources {
+		for pname, paths := range pprofPaths {
+			rname := sname + "/" + pname
```

Comment:
This should use either:
`fmt.Sprintf("%s/%s", sname, pname)`, or
`path.Join(sname, pname)`

Strings in Go are immutable and avoiding unnecessary allocations is key to performant code. Not fundamentally necessary for this PR.